DataDog/datadog-lambda-python
# bump ddtrace to take advantage of new aws sdk request span tags & reg…

### What does this PR do?
bump dd trace, regen lock file


### Motivation
SLS-3237
so we can get benefits from this release
https://github.com/DataDog/dd-trace-py/releases/tag/v1.14.0#:~:text=aws%3A%20Adds%20span%20tags%20for%20consistency%20with%20tags%20collected%20by%20Datadog%20for%20AWS%20metrics%20and%20logs.


### Testing Guidelines
Made a RC in sls sandbox in us-east-1. Specifically datadog-python38 v5 on self monitoring
Additionally this is to make sure the integration tests run on the version of ddtrace that will be utilized in the release.


### Additional Notes



### Types of Changes


### Check all that apply
